packages/terra-navigation-layout/src/NavigationLayout.jsx

```diff
@@ -0,0 +1,172 @@
+   */
+  config: navigationLayoutConfigPropType.isRequired,
+  /**
+   * The index path of the consuming application's routing structure. If provided, the NavigationLayout will
+   * ensure Redirects are present where necessary.
+   */
+  indexPath: PropTypes.string,
+  /**
+   * The location as provided by the `withRouter()` HOC.
+   */
+  location: PropTypes.object,
+  /**
+   * The match as provided by the `withRouter()` HOC.
+   */
+  match: PropTypes.object,
+  /**
+   * The history as provided by the `withRouter()` HOC.
+   */
+  history: PropTypes.object,
+  /**
+   * The staticContext as provided by the `withRouter()` HOC.
+   */
+  staticContext: PropTypes.object,
+};
+
+/**
+ * The NavigationLayout utilizes the Terra Layout and a configuration object to generate a routing-based application layout.
+ */
+class NavigationLayout extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.updateSize = this.updateSize.bind(this);
+
+    this.state = {
+      size: getBreakpointSize(),
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', this.updateSize);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateSize);
+  }
+
+  updateSize() {
+    const newSize = getBreakpointSize();
+
+    if (this.state.size !== newSize) {
+      this.setState({
+        size: newSize,
+      });
+    }
+  }
+
+  decorateElement(element, routeConfig) {
+    if (!element) {
+      return null;
+    }
+
+    const { app } = this.props;
+    const { size } = this.state;
+
+    return React.cloneElement(element, {
+      app,
+      routeConfig,
+      navigationLayoutSize: size,
+    });
+  }
+
+  render() {
+    const {
+      header,
+      children,
+      menu,
+      app,
+      menuText,
+      config,
+      indexPath,
+      location,
+      match,
+      history,
+      staticContext,
+      ...customProps
+    } = this.props; // eslint-disable-line no-unused-vars
+    const { size } = this.state;
+
+    const headerComponent = header || <NavigationLayoutContent />;
+
+    const contentComponent = children || <NavigationLayoutContent redirect={indexPath} />;
+
+    let menuComponent = menu;
+    // The configuration is examined for evidence of a valid menu component for the current size and location.
+    // If one is not found, we do not provide one to the Layout to ensure that the Layout renders appropriately.
+    if (!menuComponent && configHasMatchingRoute(location.pathname, config.menu, size)) {
+      menuComponent = <NavigationLayoutContent stackNavigationIsEnabled />;
+    }
+
+    return (
+      <Layout
+        {...customProps}
+        header={this.decorateElement(headerComponent, config.header)}
+        menu={this.decorateElement(menuComponent, config.menu)}
+        menuText={menuText}
+      >
+        {this.decorateElement(contentComponent, config.content)}
+      </Layout>
+    );
+  }
+}
+
+NavigationLayout.propTypes = propTypes;
```

Comment:
No need for defaults?

Reply:
For most, the undefined value of unprovided props is all the default we want. The config prop is marked as required, as that's really the only one that doesn't make sense to ever omit. I can mark the router-provided props as required, though that doesn't change much from a consumer standpoint.